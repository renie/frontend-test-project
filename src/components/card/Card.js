import React from 'react';
import PostList from '../posts/PostList'
import AlbumList from '../albums/AlbumList'
import './card.css';


const Card = ({user, toggleCard}) => {
  return (
    <li className={'card' + (user.isHidden ? ' hidden' : '')}>
      <header onClick={toggleCard} title="Click to toggle card" role="button">
        <h3>{user.name}</h3>
      </header>
      <div className="body">
        <dl>
          <dt>E-mail:</dt>
          <dd><a href={"mailto:" + user.email}>{user.email}</a></dd>
          <dt>Company:</dt>
          <dd>{user.company.name}</dd>
          <PostList userId={user.id} />
          <AlbumList userId={user.id} />
        </dl>
      </div>
    </li>
  );
}

export default Card;
