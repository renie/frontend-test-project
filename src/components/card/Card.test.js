import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card'
import { shallow } from 'enzyme';

describe('Card', () => {
  it('Create Component', () => {
    const user = {
      id: 10,
      name: 'John Dalton',
      email: 'j.dalton@company.com',
      company: {
        name: 'Huuuuge company'
      }
    }

    const wrapper = shallow(<Card user={user} />);

    expect(wrapper).toMatchSnapshot();
  });
});
