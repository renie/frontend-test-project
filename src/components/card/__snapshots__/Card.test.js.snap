// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Create Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    user={
      Object {
        "company": Object {
          "name": "Huuuuge company",
        },
        "email": "j.dalton@company.com",
        "id": 10,
        "name": "John Dalton",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          role="button"
          title="Click to toggle card"
        >
          <h3>
            John Dalton
          </h3>
        </header>,
        <div
          className="body"
        >
          <dl>
            <dt>
              E-mail:
            </dt>
            <dd>
              <a
                href="mailto:j.dalton@company.com"
              >
                j.dalton@company.com
              </a>
            </dd>
            <dt>
              Company:
            </dt>
            <dd>
              Huuuuge company
            </dd>
            <UNDEFINED
              userId={10}
            />
            <UNDEFINED
              userId={10}
            />
          </dl>
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            John Dalton
          </h3>,
          "onClick": undefined,
          "role": "button",
          "title": "Click to toggle card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "John Dalton",
          },
          "ref": null,
          "rendered": "John Dalton",
          "type": "h3",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <dl>
            <dt>
              E-mail:
            </dt>
            <dd>
              <a
                href="mailto:j.dalton@company.com"
              >
                j.dalton@company.com
              </a>
            </dd>
            <dt>
              Company:
            </dt>
            <dd>
              Huuuuge company
            </dd>
            <UNDEFINED
              userId={10}
            />
            <UNDEFINED
              userId={10}
            />
          </dl>,
          "className": "body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <dt>
                E-mail:
              </dt>,
              <dd>
                <a
                  href="mailto:j.dalton@company.com"
                >
                  j.dalton@company.com
                </a>
              </dd>,
              <dt>
                Company:
              </dt>,
              <dd>
                Huuuuge company
              </dd>,
              <UNDEFINED
                userId={10}
              />,
              <UNDEFINED
                userId={10}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "E-mail:",
              },
              "ref": null,
              "rendered": "E-mail:",
              "type": "dt",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="mailto:j.dalton@company.com"
                >
                  j.dalton@company.com
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "j.dalton@company.com",
                  "href": "mailto:j.dalton@company.com",
                },
                "ref": null,
                "rendered": "j.dalton@company.com",
                "type": "a",
              },
              "type": "dd",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Company:",
              },
              "ref": null,
              "rendered": "Company:",
              "type": "dt",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Huuuuge company",
              },
              "ref": null,
              "rendered": "Huuuuge company",
              "type": "dd",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "userId": 10,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(PostList)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "userId": 10,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(AlbumList)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": "dl",
        },
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            role="button"
            title="Click to toggle card"
          >
            <h3>
              John Dalton
            </h3>
          </header>,
          <div
            className="body"
          >
            <dl>
              <dt>
                E-mail:
              </dt>
              <dd>
                <a
                  href="mailto:j.dalton@company.com"
                >
                  j.dalton@company.com
                </a>
              </dd>
              <dt>
                Company:
              </dt>
              <dd>
                Huuuuge company
              </dd>
              <UNDEFINED
                userId={10}
              />
              <UNDEFINED
                userId={10}
              />
            </dl>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              John Dalton
            </h3>,
            "onClick": undefined,
            "role": "button",
            "title": "Click to toggle card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "John Dalton",
            },
            "ref": null,
            "rendered": "John Dalton",
            "type": "h3",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <dl>
              <dt>
                E-mail:
              </dt>
              <dd>
                <a
                  href="mailto:j.dalton@company.com"
                >
                  j.dalton@company.com
                </a>
              </dd>
              <dt>
                Company:
              </dt>
              <dd>
                Huuuuge company
              </dd>
              <UNDEFINED
                userId={10}
              />
              <UNDEFINED
                userId={10}
              />
            </dl>,
            "className": "body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <dt>
                  E-mail:
                </dt>,
                <dd>
                  <a
                    href="mailto:j.dalton@company.com"
                  >
                    j.dalton@company.com
                  </a>
                </dd>,
                <dt>
                  Company:
                </dt>,
                <dd>
                  Huuuuge company
                </dd>,
                <UNDEFINED
                  userId={10}
                />,
                <UNDEFINED
                  userId={10}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "E-mail:",
                },
                "ref": null,
                "rendered": "E-mail:",
                "type": "dt",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="mailto:j.dalton@company.com"
                  >
                    j.dalton@company.com
                  </a>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "j.dalton@company.com",
                    "href": "mailto:j.dalton@company.com",
                  },
                  "ref": null,
                  "rendered": "j.dalton@company.com",
                  "type": "a",
                },
                "type": "dd",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Company:",
                },
                "ref": null,
                "rendered": "Company:",
                "type": "dt",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Huuuuge company",
                },
                "ref": null,
                "rendered": "Huuuuge company",
                "type": "dd",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "userId": 10,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(PostList)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "userId": 10,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(AlbumList)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
            ],
            "type": "dl",
          },
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
